File: app/src/main/java/com/ramadan/kotlinoop/java/Book.java
package com.ramadan.kotlinoop.java;
public class Book {

    private String title;
    private String desc;
    private long isbn;

    public  Book(String title , String desc  , long isbn){
        this.title = title;
        this.desc = desc;
        this.isbn = isbn;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public long getIsbn() {
        return isbn;
    }

    public void setIsbn(long isbn) {
        this.isbn = isbn;
    }
}

File: app/src/main/java/com/ramadan/kotlinoop/kotlin/Book.kt
package com.ramadan.kotlinoop.kotlin

import java.lang.IllegalArgumentException

class Book ( var title: String,  val isbn: Long ){
    var desc : String  ="default"
    set(value) {
        if (!value.isNotEmpty()){
            throw  IllegalArgumentException("Description should not be empty !!")
        }
        field = value
    }

    get() {
        return  field.toUpperCase()
    }
}